admin/services/data-access/group.data.service.js

```diff
@@ -29,6 +30,34 @@ groupDataService.sqlFindGroups = async (schoolId) => {
   return sqlService.query(sql, params)
 }
 
+/**
+ * Get active groups (non-soft-deleted), which have at least one present pupil.
+ * @param schoolId
+ * @returns {Promise<*>}
+ */
+groupDataService.sqlFindGroupsWithAtleastOnePresentPupil = async (schoolId) => {
+  const sql = `
+  SELECT g.id, g.name, COUNT(pg.pupil_id) as pupilCount 
+  FROM ${sqlService.adminSchema}.[group] g
+  LEFT OUTER JOIN ${sqlService.adminSchema}.pupilGroup pg 
+  ON g.id = pg.group_id
+  LEFT JOIN ${sqlService.adminSchema}.pupilAttendance pa
+  ON pa.pupil_id=pg.pupil_id
+  WHERE pa.id IS NULL
+  AND g.isDeleted=0
+  AND g.school_id=@schoolId
+  GROUP BY g.id, g.name
+  ORDER BY name ASC`
```

Comment:
I wonder whether it wouldn't be better to order the results in the application servers rather than the DB.  There is only 1 DB server, but we can horizontally scale the app servers.

This comment is not specific to this code but rather would apply across the app.

Reply:
Having looked at this, I am of the opinion that moving the ordering to the application will have a negligible effect. I think also that the work being done elsewhere to improve the performance of the data layer will have a larger impact.